src/main/kotlin/lotto/domain/ticket/WinningBoard.kt

```diff
@@ -8,13 +8,21 @@ enum class WinningBoard(
     THREE(3, 5_000),
     FOUR(4, 50_000),
     FIVE(5, 150_000),
+    FIVE_WITH_BONUS(5, 30_000_000),
     SIX(6, 2_000_000_000),
     ;
 
-    fun times(amount:Int): Long {
+    fun times(amount: Int): Long {
         return reward.times(amount)
     }
 
+    fun upgrade(isBonusMatch: Boolean) =
+        if (isBonusMatch && this == FIVE) {
+            FIVE_WITH_BONUS
+        } else {
+            this
+        }
+
     companion object {
         fun findBy(matchCount: Int) = values().find { it.matchCount == matchCount }
```

Comment:
```suggestion
         fun findBy(matchCount: Int, bonusMatch: Boolean): WinningBoard {
            if(matchCount == FIVE_WITH_BONUS.matchCount && bonusMatch) {
                return FIVE_WITH_BONUS
            }

            return values().find { it.matchCount == matchCount } ?: NONE
        }
```
이렇게 표현도 가능합니다.
참고만해주세요 :)